src/Cardano/Signer.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Signer.h"
+#include "Address.h"
+#include "../Cbor.h"
+#include "../Data.h"
+#include "../HexCoding.h"
+#include "../PrivateKey.h"
+#include "../PublicKey.h"
+
+using namespace TW;
+using namespace TW::Cardano;
+using namespace std;
+
+
+Proto::SigningOutput Signer::sign(const Proto::SigningInput& input) {
+    Data txId;
+    Data signedTxEncoded = prepareSignedTx(input, txId);
+    Proto::SigningOutput output;
+    //output.set_transaction // TODO
+    output.set_encoded(signedTxEncoded.data(), signedTxEncoded.size());
+    output.set_transaction_id(hex(txId));
+    output.set_fee(input.fee());
+    return output;
+}
+
+Data Signer::prepareUnsignedTx(const Proto::SigningInput& input) {
+    // inputs array
+    uint64_t sum_utxo = 0;
+    auto inputsArray = Cbor::Encode::indefArray();
```

Comment:
```suggestion
    auto inputsArray = Encode::indefArray();
```
do you consider to `using namespace CBor`? namespaces more than two levels looks verbose

Reply:
ok